src/finitio/type/tuple_type.coffee

```diff
@@ -47,6 +47,15 @@ class TupleType extends Type
         m.onFailure (f)->
           f.location = attrName
           m
+      else if !attr? and @heading.allowExtra()
+        type = @heading.getExtraType()
+        subm = type.mDress(attrValue, Monad)
+        subm.onFailure (error)->
+          error.location = attrName
+          subm
+        subm.onSuccess (val)->
+          result[attrName] = val
+          success
```

Comment:
Some cases later in the if/then/else might end up being unnecessary and must be removed/cleaned.